### PR TITLE
add choose_devices()

### PR DESCRIPTION
**Pull request by matthiasdiener**

### CODE REVIEW

File: pyopencl/__init__.py [hunk 1]
```diff
@@ -1576,7 +1577,7 @@ def create_some_context(interactive: Optional[bool] = None,
             from pyopencl.tools import get_test_platforms_and_devices
             for _plat, devs in get_test_platforms_and_devices():
                 for dev in devs:
-                    return Context([dev])
```
Comment (matthiasdiener) [inline pyopencl/__init__.py:1579]: This code looks a bit strange, why not just return `Context([devs[0]])`?

Comment (alexfikl) [reply pyopencl/__init__.py:1579]: That wouldn't work if `devs` is empty?

Comment (matthiasdiener) [reply pyopencl/__init__.py:1579]: Oh, I see. Is that a Python pattern? I would just have written `if devs: return ...`

Comment (alexfikl) [reply pyopencl/__init__.py:1579]: I don't think it's a common pattern, but it avoids an `if` in Python. Not sure if that's much faster + this isn't a hot path..

File: pyopencl/__init__.py [hunk 2]
```diff
@@ -1552,10 +1552,10 @@ def __init__(self, ctx, size, alignment, flags, *, _interface,
 
 # {{{ create_some_context
 
-def create_some_context(interactive: Optional[bool] = None,
-                        answers: Optional[List[str]] = None) -> Context:
+def choose_device(interactive: Optional[bool] = None,
+                  answers: Optional[List[str]] = None) -> List[Device]:
```
Comment (alexfikl) [inline pyopencl/__init__.py]: Naming wise, it's weird that this returns a list and it's called `choose_device`?

Comment (matthiasdiener) [reply pyopencl/__init__.py]: Yeah, we could rename it to `choose_devices` or so. My only concern was that most (?) people will just choose 1 device...

**Pull request merged**